eq-author-api/schema/tests/duplication.test.js

```diff
@@ -0,0 +1,158 @@
+    section = last(questionnaire.sections);
+  });
+
+  afterAll(async () => {
+    await deleteQuestionnaire(questionnaire.id);
+  });
+
+  describe("duplicate a page", async () => {
+    let page, pageCopy;
+
+    beforeEach(async () => {
+      page = await queryQuestionPage(questionnaire, last(section.pages).id);
+      let { id } = await duplicatePage(questionnaire, page);
+      pageCopy = await queryQuestionPage(questionnaire, id);
+    });
+
+    afterEach(async () => {
+      await deleteQuestionPage(questionnaire, pageCopy.id);
+    });
+
+    it("should copy page with answers and question confirmation", () => {
+      expect(pageCopy).toEqual(
+        expect.objectContaining(
+          omit(page, ["id", "alias", "title", "displayName", "position"])
+        )
+      );
+    });
+
+    it("should have new id", () => {
+      expect(pageCopy.id).not.toEqual(page.id);
+    });
+
+    it("should create new title", () => {
+      expect(pageCopy.title).toEqual(`Copy of ${page.title}`);
+    });
+
+    it("should correctly increment position", () => {
+      expect(pageCopy.position).toEqual(page.position + 1);
+    });
+  });
+
+  describe("duplicate a section", async () => {
+    let queriedSection;
+    let sectionCopy;
+
+    beforeEach(async () => {
+      queriedSection = await querySection(questionnaire, section.id);
+      let { id } = await duplicateSection(questionnaire, queriedSection);
+      sectionCopy = await querySection(questionnaire, id);
+    });
+
+    afterEach(async () => {
+      await deleteSection(questionnaire, sectionCopy.id);
+    });
+
+    it("should copy section with pages", () => {
+      expect(sectionCopy).toEqual(
+        expect.objectContaining(
+          omit(queriedSection, [
+            "id",
+            "alias",
+            "title",
+            "displayName",
+            "position",
+          ])
+        )
+      );
+    });
+
+    it("should have new id", () => {
+      expect(sectionCopy.id).not.toEqual(queriedSection.id);
+    });
+
+    it("should create new title", () => {
+      expect(sectionCopy.title).toEqual(`Copy of ${queriedSection.title}`);
+    });
+
+    it("should correctly increment position", async () => {
+      expect(sectionCopy.position).toEqual(queriedSection.position + 1);
+    });
+  });
+
+  describe("duplicate a questionnaire", async () => {
+    let queriedQuestionnaire;
+    let questionnaireCopy;
+    let duplicatedQuestionnaire;
+
+    beforeEach(async () => {
+      queriedQuestionnaire = await queryQuestionnaire(questionnaire);
+      duplicatedQuestionnaire = await duplicateQuestionnaire(questionnaire);
+      questionnaireCopy = await queryQuestionnaire(duplicatedQuestionnaire);
+    });
+
+    afterEach(async () => {
+      await deleteQuestionnaire(questionnaireCopy.id);
+    });
+
+    it("should copy questionnaire with sections and pages", () => {
+      //@todo - Some sort of issue with createdBy resolving correctly
```

Comment:
I fixed this so that it would work for create questionnaire. You can see how that works. The auth value is currently not configurable though